pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -0,0 +1,279 @@
+		/// A bonded token pool has been moved to destroying state. [pool_id]
+		DestructionStarted(T::AccountId),
+		/// A bonded token pool has been fully destroyed and all collateral and deposits have been refunded. [pool_id]
+		Destroyed(T::AccountId),
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The number of bonded currencies on a new pool is either lower than 1 or greater than MaxCurrencies.
+		CurrenciesOutOfBounds,
+		/// A token swap cannot be executed due to a lock placed on this operation.
+		Locked,
+		/// The pool id is not currently registered.
+		PoolUnknown,
+		/// The pool has no associated bonded currency with the given index.
+		IndexOutOfBounds,
+		/// The cost or returns for a mint, burn, or swap operation is outside the user-defined slippage tolerance.
+		Slippage,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		FungiblesBalanceOf<T>: TryInto<CollateralCurrencyBalanceOf<T>>,
+	{
+		#[pallet::call_index(0)]
+		// #[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))] TODO: properly configure weights
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: Curve,
+			currencies: BoundedVec<TokenMeta<FungiblesBalanceOf<T>, FungiblesAssetIdOf<T>>, T::MaxCurrencies>,
+			frozen: bool,
+			// currency_admin: Option<AccountIdLookupOf<T>> TODO: use this to set currency admin
+		) -> DispatchResultWithPostInfo {
+			// ensure origin is PoolCreateOrigin
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
+
+			ensure!(
+				(2..=T::MaxCurrencies::get().saturated_into()).contains(&currencies.len()),
+				Error::<T>::CurrenciesOutOfBounds
+			);
+
+			let currency_ids = BoundedVec::truncate_from(currencies.iter().map(|c| c.id.clone()).collect());
+
+			let pool_id = T::PoolId::from(currency_ids.blake2_256());
+
+			T::DepositCurrency::hold(
+				&pool_id.clone().into(), // TODO: just assumed that you can use a pool id as hold reason, not sure that's true though
+				&who,
+				T::DepositPerCurrency::get()
+					.saturating_mul(currencies.len().saturated_into())
+					.saturated_into(),
+			)?;
+
+			for entry in currencies {
+				let asset_id = entry.id.clone();
+
+				// create new assset class; fail if it already exists
+				T::Fungibles::create(asset_id.clone(), pool_id.clone().into(), false, entry.min_balance)?;
+
+				// set metadata for new asset class
+				T::Fungibles::set(
+					asset_id,
+					&pool_id.clone().into(),
+					entry.name.clone(),
+					entry.symbol.clone(),
+					entry.decimals,
+				)?;
+
+				// TODO: use fungibles::roles::ResetTeam to update currency admin
+			}
+
+			<Pools<T>>::set(
+				pool_id,
+				Some(PoolDetails::new(who.clone(), curve, currency_ids, !frozen)),
+			);
+
+			// Emit an event.
+			Self::deposit_event(Event::PoolCreated(who));
+			// Return a successful DispatchResultWithPostInfo
+			Ok(().into())
```

Comment:
If you change the function signature to `DispatchResult`, you don't need the into anymore

Reply:
your call I'd say, if this is what we use everywhere we should probably do this here too